test/functional/test_framework/address.py

```diff
@@ -55,17 +55,15 @@ def create_deterministic_address_bcrt1_p2tr_op_true():
 
 def byte_to_base58(b, version):
     result = ''
-    str = b.hex()
-    str = chr(version).encode('latin-1').hex() + str
-    checksum = hash256(bytes.fromhex(str)).hex()
-    str += checksum[:8]
-    value = int('0x' + str, 0)
+    b = bytes([version]) + b  # prepend version
+    b += hash256(b)[:4]       # append checksum
+    value = int.from_bytes(b, 'big')
     while value > 0:
         result = chars[value % 58] + result
         value //= 58
-    while (str[:2] == '00'):
+    while (b[0] == 0):
```

Comment:
Might as well remove the unnecessary parentheses here while we're at it.

Reply:
Thanks, good catch, done.